sql/core/src/test/scala/org/apache/spark/sql/StatisticsSuite.scala

```diff
@@ -101,4 +101,47 @@ class StatisticsSuite extends QueryTest with SharedSQLContext {
       checkTableStats(tableName, expectedRowCount = Some(2))
     }
   }
+
+  test("test column-level statistics for data source table created in InMemoryCatalog") {
+    def checkColStats(colStats: BasicColStats, expectedColStats: BasicColStats): Unit = {
+      assert(colStats.dataType == expectedColStats.dataType)
+      assert(colStats.numNulls == expectedColStats.numNulls)
+      assert(colStats.max == expectedColStats.max)
+      assert(colStats.min == expectedColStats.min)
+      if (expectedColStats.ndv.isDefined) {
+        // ndv is an approximate value, so we just make sure we have the value
+        assert(colStats.ndv.get >= 0)
```

Comment:
`assert(colStats.ndv.get >= 0)` might be too loose. Can we check if it is more than 0 and less than the number of values in the column?